src/Support/Middleware.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rebing\GraphQL\Support;
+
+use Closure;
+use GraphQL\Type\Definition\ResolveInfo;
+
+abstract class Middleware
+{
+    public function handle($root, $args, $context, ResolveInfo $info, Closure $next)
+    {
+        return $next($root, $args, $context, $info);
+    }
+
+    public function resolve(array $arguments, Closure $next)
```

Comment:
Please add a phpdoc that this is called from `\Rebing\GraphQL\Support\Field::getResolver` (no one can statically analyze due to how the code has to work with the string name of this method :/)

Reply:
Something like this?

```suggestion
    /**
     * @see Field::getResolver()  Middleware is resolved in the field resolver pipeline
     */
    public function resolve(array $arguments, Closure $next)
```

Reply:
A related thing I considered was swapping the `handle` and `resolve` names around i.e. so the base class has `handle` that calls `resolve` instead of the current implementation `resolve` that calls `handle` – the default `via` pipelines use is `handle`.

I did it this way round so that end-user GraphQL middleware are more consistent with HTTP (by having a `handle` method) but I'm not sure if that's as useful as them having a `resolve` method to make it clear they intercept resolution rather than the HTTP stack.

Reply:
TBH that thought didn't cross my mind.

In fact it felt "natural" that it's the `handle` method one has to implement 🤷‍♀️

Reply:
Yeah, the `@see` seems appropriate!

Reply:
Handle makes sense to me too, I am happy to run with that